ek9000App/src/devEK9000.cpp

```diff
@@ -422,13 +403,25 @@ int devEK9000::InitTerminal(int term) {
 /* This will configure process image locations in each terminal */
 /* It will also verify that terminals have the correct type (reads terminal type from the device then yells if its not
 the same as the user specified.) */
-int devEK9000::InitTerminals() {
+bool devEK9000::ComputeTerminalMapping() {
 	if (m_init) {
 		epicsPrintf("devEK9000: Already initialized.\n");
-		return 1;
+		return false;
 	}
 	m_init = true;
 
+	/* Gather a buffer of connected terminals */
+	uint16_t railLayout[0xFF];
+	memset(railLayout, 0, sizeof(railLayout));
+	for (int i = 0; i < 0xFF; i += 64) {
+		if (this->doModbusIO(0, MODBUS_READ_HOLDING_REGISTERS, 0x6001 + i, railLayout + i, 64) != asynSuccess) {
+			epicsPrintf("%s: Failed to read rail layout from the device\n", __FUNCTION__);
+			return false;
+		}
+	}
+
+	assert(m_numTerms <= ArraySize(railLayout));
```

Comment:
This is probably worth a real error message before exiting?

Reply:
I've clamped this in `ek9000Configure` now, no user-facing error message is needed in here. leaving the assert just for my own sanity.